### PR TITLE
FORMS-8863 containsFormContainer API to check if resource contains ad…

### PR DESCRIPTION
**Pull request by barshat7**
…aptive form container

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormStructureParser.java [hunk 1]
```diff
@@ -37,4 +37,9 @@ public interface FormStructureParser {
      */
     String getClientLibRefFromFormContainer();
 
+    /**
+     *
+     * @return true if this resource contains a form container
+     */
+    Boolean containsFormContainer();
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormStructureParser.java:44]: Can you improve the function documentation ?

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormStructureParserImpl.java [hunk 2]
```diff
@@ -43,6 +43,26 @@ public String getClientLibRefFromFormContainer() {
         return getPropertyFromFormContainer(resource, FormContainer.PN_CLIENT_LIB_REF);
     }
 
+    @Override
+    public Boolean containsFormContainer() {
+        return containsFormContainer(resource);
+    }
+
+    private Boolean containsFormContainer(Resource resource) {
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormStructureParserImpl.java:51]: Why did you create another method, it is confusing. Can you add it the original method ?

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormStructureParserImpl.java (hunk 2 above)
Comment (barshat7) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormStructureParserImpl.java:51]: We'll need to recursively traverse the children nodes to check whether there exists a resource with fieldType is form. Its a private recursive method.

**Pull request merged**